SBR - s3 Branch

s3anyboy opened the pull request:
1.0.1
Changed "en-US" to simply "en" for better compatibility, also added en.SDH as language option for SDH/HI subs
Added support for every existing 2 letter ISO 639 language code
Added Icon
Changed all application titles from SubtitleRenamer to SubtitleBatchRenamer to match the repo (and help avoid confusion since there are already a ton of projects named simply SubtitleRenamer)

s3anyboy committed:
Update README.MD

s3anyboy committed:
Update MainForm.Designer.cs

added en.SDH for SDH/HI subs, changes en-us to en for better compatibility, added all current (2024) valid ISO 639 language codes

s3anyboy committed:
Update MainForm.Designer.cs

SubtitleRenamer to SubtitleBatchRenamer

s3anyboy committed:
Update Program.cs

s3anyboy committed:
Update MainForm.cs

s3anyboy committed:
Update SubtitleRenamer.sln

s3anyboy committed:
Rename SubtitleRenamer.sln to SubtitleBatchRenamer.sln

s3anyboy committed:
Rename SubtitleRenamer.csproj to SubtitleBatchRenamer.csproj

s3anyboy committed:
Rename + Icon

Changed all instances of SubtitleRenamer to SubtitleBatchRenamer
Added Icon

s3anyboy committed:
Update README.MD

jameswsullivan commented:
Hi, thanks for your updates to the repo.

I have a question regarding the `en-US` vs. `en` though: when I first wrote it that way, I was thinking about whether to accommodate these: `en-au`, `en-bg`, `en-us`, `en-ca`, etc., like the ones listed [here](https://www.andiamo.co.uk/resources/iso-language-codes/), or like the ones you'd choose/configure when configuring Windows' keyboard layout.

I personally haven't run into this, but thought if it'd be helpful if the tool is used by a subtitle group or something that might deal with different language variants, such as `zh-cn`, `zh-tw` which would be simplified vs. traditional.

I think adding these common variants would be a better idea?

jameswsullivan commented:
Oh, also, I think maybe adding a CHANGELOG.MD to document changes  made by each contributor?

s3anyboy commented:
> Hi, thanks for your updates to the repo.
> 
> I have a question regarding the `en-US` vs. `en` though: when I first wrote it that way, I was thinking about whether to accommodate these: `en-au`, `en-bg`, `en-us`, `en-ca`, etc., like the ones listed [here](https://www.andiamo.co.uk/resources/iso-language-codes/), or like the ones you'd choose/configure when configuring Windows' keyboard layout.
> 
> I personally haven't run into this, but thought if it'd be helpful if the tool is used by a subtitle group or something that might deal with different language variants, such as `zh-cn`, `zh-tw` which would be simplified vs. traditional.
> 
> I think adding these common variants would be a better idea?

I'm not sure, I just know Bazarr (a subtitle management program) had issues recognizing the en-US files, not entirely sure how Chinese Simplified vs Traditional would be processed.

CHANGELOG is also a good idea

jameswsullivan commented:
> Bazarr (a subtitle management program) had issues recognizing the en-US files

Did you have insights as to how/why, such as having trouble with the dashes or something deeper like encoding-wise?

s3anyboy committed:
Major Version 1.1.0

Major Version 1.1.0

Fixed MainForm so it can be opened in designer again (in VS22 at least)
Added About button/window
Added "Add Label" and "Label" columns for adding additional labels (SDH, FORCED, etc)
Moved language list to language.txt so users can easily edit it to add/remove languages as needed
Slightly better icon

s3anyboy committed:
Delete App.config

s3anyboy commented:
> > Bazarr (a subtitle management program) had issues recognizing the en-US files
> 
> Did you have insights as to how/why, such as having trouble with the dashes or something deeper like encoding-wise?

Seems like it was actually probably just something with my Bazarr setup, but after looking into the most popular media players I came up with what I think is a good solution, I offloaded the languages list to external language.txt and while I included pretty much every language I think someone might reasonably use (including some silly ones like Klingon and Quenya), and some of the most common (and substantial) language variant country codes, but this way an end user can easily add/remove languages from the list as needed.

Made some other changes too (added additional label column instead of having .SDH language variants), I'd like to make a changelog but I'm not entirely sure how to go about formatting it?

s3anyboy committed:
MINOR 1.1.1

Fixed error and omission in about box

s3anyboy committed:
Update README.MD

s3anyboy committed:
Update README.MD

s3anyboy committed:
Rename LICENSE.md to LICENSE.MD

cuz it looks better lol

s3anyboy commented on MainForm.Designer.cs:
Also added support for additional video file types in 1.1.0 (FLV, MOV, MPEG/MPG, M4V and WEBM)

s3anyboy committed:
1.1.2 MINOR UI

Enabled alternating colors for rows for better readability.
Fixed About button not being properly anchored.
Fixed Add Label Check All button not being properly disabled when unavailable.
Minor UI tweaks

s3anyboy committed:
Update MainForm.cs

 DataGridViewCheckBoxCell

s3anyboy committed:
1.1.3 MINOR

Code cleanup

s3anyboy committed:
Update SubtitleBatchRenamer.csproj

s3anyboy committed:
Update README.MD

updated readme to match about

s3anyboy committed:
Update README.MD

s3anyboy committed:
Delete about.txt

s3anyboy committed:
Add CHANGELOG.MD

Add CHANGELOG

jameswsullivan commented:
Hey,

I had thought about adding a "config file" for the a list of languages that allows users to add their own. However, on a second thought I was like "what if odd characters are added and my app doesn't know how to parse it and create unpredictable edge cases...", that's why I ended up not doing it that way.

What if we do it this way, I think it's better practice:

- We create a List of some sort in the [application settings](https://learn.microsoft.com/en-us/visualstudio/ide/managing-application-settings-dotnet?view=vs-2022) (I'm drawing blank on the proper name/type of it at this moment, like a List of strings),
- define all the language codes in a class or something, and load these default ones into the Settings upon app initialization,
- also allow users to add their own language codes, and save it in the Settings,
- before saving to Settings, we add a validation so that we only allow: a certain length, no weird characters, and normalize it before saving, otherwise prompt an error/message saying "please name it _this way_".

But I'll need to play around with my ideas a bit to remember how to write these, let me create a [language-codes branch](https://github.com/jameswsullivan/SubtitleBatchRenamer/tree/language-codes) for us to merge our ideas into before directly merging to main.


As for CHANGELOG format, I'm thinking something [along this line](https://github.com/jameswsullivan/open-oni/blob/customizations/CUSTOMIZATIONS.md) would be fine, and add our names to each section of the changes. What you already did also looks fine.

s3anyboy commented:
This is a good point I hadn't really considered, I guess it would probably be good to also maybe put file types and labels in a similar settings menu, sup that way if someone is using some obscure old (or futuristic new) video file types they can easily add support for them (and maybe subtitle file types list too?)

jameswsullivan commented:
Yeah these are all good thinking. Let's make a to-do list of some sort and add one improvement at a time.
